CalendarSync exception

The following exception was reported via Google:

````
java.lang.RuntimeException: Unable to create application nl.mpcjanssen.simpletask.TodoApplication: java.lang.IllegalArgumentException: Unknown URL content://com.android.calendar/calendars?caller_is_syncadapter=true&account_name=Simpletask%20Calendar&account_type=LOCAL
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4596)
	at de.robv.android.xposed.XposedBridge.invokeOriginalMethodNative(Native Method)
	at de.robv.android.xposed.XposedBridge.handleHookedMethod(XposedBridge.java:668)
	at android.app.ActivityThread.handleBindApplication(<Xposed>)
	at android.app.ActivityThread.access$1600(ActivityThread.java:154)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1382)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5294)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:115)
Caused by: java.lang.IllegalArgumentException: Unknown URL content://com.android.calendar/calendars?caller_is_syncadapter=true&account_name=Simpletask%20Calendar&account_type=LOCAL
	at android.content.ContentResolver.insert(ContentResolver.java:1203)
	at nl.mpcjanssen.simpletask.CalendarSync.addCalendar(CalendarSync.java:118)
	at nl.mpcjanssen.simpletask.CalendarSync.setSyncType(CalendarSync.java:242)
	at nl.mpcjanssen.simpletask.CalendarSync.<init>(CalendarSync.java:264)
	at nl.mpcjanssen.simpletask.TodoApplication.onCreate(TodoApplication.java:113)
	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1012)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4593)
	... 13 more
````

--- a/src/main/java/nl/mpcjanssen/simpletask/CalendarSync.java
+++ b/src/main/java/nl/mpcjanssen/simpletask/CalendarSync.java
@@ -86,13 +86,18 @@ public class CalendarSync {
         final String[] projection = {Calendars._ID, Calendars.NAME};
         final String selection = Calendars.NAME+" = ?";
         final String[] args = {CAL_NAME};
-        Cursor cursor = m_cr.query(CAL_URI, projection, selection, args, null);
-        if (cursor == null) return -1;
-        if (cursor.getCount() == 0) return -1;
-        cursor.moveToFirst();
-        long ret = cursor.getLong(0);
-        cursor.close();
-        return ret;
+        try {
+            Cursor cursor = m_cr.query(CAL_URI, projection, selection, args, null);
+            if (cursor == null) return -1;
+            if (cursor.getCount() == 0) return -1;
+            cursor.moveToFirst();
+            long ret = cursor.getLong(0);
+            cursor.close();
+            return ret;
+        }
+        catch (IllegalArgumentException e) {
+            return -1;
+        }
     }
 
     private void addCalendar(boolean warnCalExists) {
@@ -115,10 +120,15 @@ public class CalendarSync {
         cv.put(Calendars.VISIBLE, 1);
         cv.put(Calendars.SYNC_EVENTS, 1);
 
-        Uri calUri = m_cr.insert(CAL_URI, cv);
         boolean added = true;
-        if (calUri == null) added = false;
-        else if (getCalID() == -1) added = false;    // This might happen eg. because of CM's privacy guard
+        try {
+            Uri calUri = m_cr.insert(CAL_URI, cv);
+            if (calUri == null) added = false;
+            else if (getCalID() == -1) added = false;    // This might happen eg. because of CM's privacy guard
+        }
+        catch (IllegalArgumentException e) {
+            added = false;
+        }
 
         if (added) {
             log.debug("Calendar added");
@@ -126,15 +136,21 @@ public class CalendarSync {
         else {
             log.error("Could not add calendar");
             m_sync_type = 0;
+            Util.showToastShort(TodoApplication.getAppContext(), R.string.calendar_add_error);
         }
     }
 
     private void removeCalendar() {
         final String selection = Calendars.NAME+" = ?";
         final String[] args = {CAL_NAME};
-        int ret = m_cr.delete(CAL_URI, selection, args);
-        if (ret == 1) log.debug("Calendar removed");
-        else log.warn("Unexpected return value while removing calendar: " + ret);
+        try {
+            int ret = m_cr.delete(CAL_URI, selection, args);
+            if (ret == 1) log.debug("Calendar removed");
+            else log.warn("Unexpected return value while removing calendar: " + ret);
+        }
+        catch (IllegalArgumentException e) {
+            log.warn("Exception while removing calendar", e);
+        }
     }
 
     @SuppressWarnings("NewApi")

--- a/src/main/res/values-de/strings.xml
+++ b/src/main/res/values-de/strings.xml
@@ -347,7 +347,7 @@ You should have received a copy of the GNU General Public License along with Tod
     <string name="calendar_sync_desc_due">Erinnerung an das Fälligkeitsdatum, automatisch erstellt von Simpletask</string>
     <string name="calendar_sync_desc_thre">Erinnerung an der Anfangsdatum, automatisch erstellt von Simpletask</string>
     <string name="calendar_exists_warning">Warnung: ein Kalender identischen Namens existiert bereits. Überschreibe...</string>
-    <string name="calendar_add_error">Fehler: Kann nicht einen Kalendar zufügen</string>
+    <string name="calendar_add_error">Fehler: Kann Kalender nicht hinzufügen</string>
 
  <string name="click_to_copy">Tippen, um in die Zwischenablage zu kopieren.</string>
 </resources>

--- a/src/main/res/values-de/strings.xml
+++ b/src/main/res/values-de/strings.xml
@@ -347,6 +347,7 @@ You should have received a copy of the GNU General Public License along with Tod
     <string name="calendar_sync_desc_due">Erinnerung an das Fälligkeitsdatum, automatisch erstellt von Simpletask</string>
     <string name="calendar_sync_desc_thre">Erinnerung an der Anfangsdatum, automatisch erstellt von Simpletask</string>
     <string name="calendar_exists_warning">Warnung: ein Kalender identischen Namens existiert bereits. Überschreibe...</string>
+    <string name="calendar_add_error">Fehler: Kann nicht einen Kalendar zufügen</string>
 
  <string name="click_to_copy">Tippen, um in die Zwischenablage zu kopieren.</string>
 </resources>

--- a/src/main/res/values-es/strings.xml
+++ b/src/main/res/values-es/strings.xml
@@ -215,6 +215,7 @@ Cualquier cambio que no haya sido sincronizado se perderá al abrir un archivo d
 <string name="calendar_sync_desc_due">Due date reminder, generated automatically by Simpletask</string>
 <string name="calendar_sync_desc_thre">Threshold date reminder, generated automatically by Simpletask</string>
 <string name="calendar_exists_warning">Aviso: Ya existe un calendario con el mismo nombre, sobreescribiendolo…</string>
+<string name="calendar_add_error">Error: Could not add calendar</string>
 
 <string name="click_to_copy">Clic para copiar al portapapeles.</string>
 

--- a/src/main/res/values-it/strings.xml
+++ b/src/main/res/values-it/strings.xml
@@ -337,6 +337,7 @@ You should have received a copy of the GNU General Public License along with Tod
     <string name="calendar_sync_desc_due">Due date reminder, generated automatically by Simpletask</string>
     <string name="calendar_sync_desc_thre">Threshold date reminder, generated automatically by Simpletask</string>
     <string name="calendar_exists_warning">Warning: Calendar with the same name already exists, overwriting...</string>
+    <string name="calendar_add_error">Error: Could not add calendar</string>
 
     <string name="click_to_copy">Click to copy to clipboard.</string>
 </resources>

--- a/src/main/res/values-ko/strings.xml
+++ b/src/main/res/values-ko/strings.xml
@@ -304,6 +304,7 @@
     <string name="calendar_sync_desc_due">Due date reminder, generated automatically by Simpletask</string>
     <string name="calendar_sync_desc_thre">Threshold date reminder, generated automatically by Simpletask</string>
     <string name="calendar_exists_warning">경고: 같은 이름의 달력이 이미 존재함, 덮어쓰는중…</string>
+    <string name="calendar_add_error">Error: Could not add calendar</string>
 
     <string name="click_to_copy">클립보드에 복사하려면 선택</string>
 </resources>

--- a/src/main/res/values/strings.xml
+++ b/src/main/res/values/strings.xml
@@ -377,6 +377,7 @@ You should have received a copy of the GNU General Public License along with Tod
     <string name="calendar_sync_desc_due">Due date reminder, generated automatically by Simpletask</string>
     <string name="calendar_sync_desc_thre">Threshold date reminder, generated automatically by Simpletask</string>
     <string name="calendar_exists_warning">Warning: Calendar with the same name already exists, overwriting...</string>
+    <string name="calendar_add_error">Error: Could not add calendar</string>
 
     <string name="click_to_copy">Click to copy to clipboard.</string>
 

